aws/request/timeout_read_closer_test.go

```diff
@@ -0,0 +1,55 @@
+package request
+
+import (
+	"io"
+	"testing"
+	"time"
+)
+
+type testReader struct {
+	duration time.Duration
+}
+
+func (r *testReader) Read(b []byte) (int, error) {
+	time.Sleep(r.duration)
+	return 0, io.EOF
+}
+
+func (r *testReader) Close() error {
+	return nil
+}
+
+func TestTimeoutReadCloser(t *testing.T) {
+	reader := timeoutReadCloser{
+		reader: &testReader{
+			duration: time.Second,
+		},
+		duration: time.Millisecond,
+	}
+	b := make([]byte, 100)
+	_, err := reader.Read(b)
+	if err != nil {
+		t.Log(err)
+	}
+}
+
+func TestTimeoutReadCloserSameDuration(t *testing.T) {
+	reader := timeoutReadCloser{
+		reader: &testReader{
+			duration: time.Millisecond,
+		},
+		duration: time.Millisecond,
+	}
+	b := make([]byte, 100)
+	_, err := reader.Read(b)
+	if err != nil {
+		t.Log(err)
+	}
+}
+
+func TestWithResponseReadTimeout(t *testing.T) {
```

Comment:
This doesn't really test anything. I suggest a test that creates a request, applies the read timeout option, and verifies that the http#Response.Body was updated correctly.